README.md

```diff
@@ -12,96 +12,123 @@ dataset), and use two types of models to predict the correct answer.
  ## Setup environment
  1. Create the `arc_solvers` environment using Anaconda
  
-   ```
-   conda create -n arc_solvers python=3.6
-   ```
+```bash
+conda create -n arc_solvers python=3.6
+```
  
  2. Activate the environment
  
-   ```
-   source activate arc_solvers
-   ```
+```bash
+source activate arc_solvers
+```
  
  3. Install the requirements in the environment: 
  
-   ```
-   sh scripts/install_requirements.sh
-   ```
+```bash
+sh scripts/install_requirements.sh
+```
  
  4. Install pytorch as per instructions on <http://pytorch.org/>. Command as of Feb. 26, 2018:
  
-   ```
-   conda install pytorch torchvision -c pytorch
-   ```
+```bash
+conda install pytorch torchvision -c pytorch
+```
   
 
  ## Setup data/models
  1. Download the data and models into `data/` folder. This will also build the ElasticSearch
  index (assumes ElasticSearch 6+ is running on `ES_HOST` machine defined in the script)
-  ```
-  sh scripts/download_data.sh
-  ```
-  
- 
- ## Running baseline models
- Run the entailment-based baseline solvers against a question set using `scripts/evaluate_solver.sh`
+```bash
+sh scripts/download_data.sh
+```
 
- For example, to evaluate the DGEM model on the Challenge Set, run:
+ 2. Download and prepare embeddings. This will download glove.840B.300d.zip from https://nlp.stanford.edu/projects/glove/ and 
+ convert it to glove.840B.300d.txt.gz which is readable from AllenNLP
+```bash
+sh download_and_prepare_glove.sh
 ```
+
+ 
+## Running baseline models
+Run the entailment-based baseline solvers against a question set using `scripts/evaluate_solver.sh`
+
+### Running a pre-trained DGEM model
+For example, to evaluate the DGEM model on the Challenge Set, run:
+```bash
 sh scripts/evaluate_solver.sh \
 	data/ARC-V1-Feb2018/ARC-Challenge/ARC-Challenge-Test.jsonl \
-	data/ARC-V1-Models-Feb2018/dgem/
+	data/ARC-V1-Models-Aug2018/dgem/
 ```
-  Change `dgem` to `decompatt` to test the Decomposable Attention model.
+Change `dgem` to `decompatt` to test the Decomposable Attention model.
 
- To evaluate the BiDAF model, use the `evaluate_bidaf.sh` script
-```
+### Running a pre-trained BiDAF model
+To evaluate the BiDAF model, use the `evaluate_bidaf.sh` script
+```bash
  sh scripts/evaluate_bidaf.sh \
     data/ARC-V1-Feb2018/ARC-Challenge/ARC-Challenge-Test.jsonl \
-    data/ARC-V1-Models-Feb2018/bidaf/
+    data/ARC-V1-Models-Aug2018/bidaf/
 ```
 
- 
- ## Running against a new question set
+### Training and evaluating the BiLSTM Max-out with Question to Choices Max Attention
+This model implements an attention interaction between question and choices context-encoded representations. The model is described [here](arc_solvers/models/qa/README.md#bilstm-max-out-with-question-to-choices-max-attention).
```

Comment:
Minor edit for readability: "... attention interaction between the context-encoded representations of the question and the choices."